frontend/workflows/projectSelector/src/project-selector.tsx

```diff
@@ -114,6 +114,10 @@ const ProjectSelector = ({ children }) => {
     }
   }, [state[Group.PROJECTS]]);
 
+  // computes the final state for rendering across other components
+  // (ie. filters out upstream/downstreams that are "hidden")
+  const derivedState = React.useMemo(() => deriveStateData(state), [state]);
```

Comment:
some perf benefits to using `useMemo` for example, without it we'd be calling this func every time the user types in the text field